tests/tensorflow/common.libsonnet

```diff
@@ -41,6 +41,49 @@ local volumes = import 'templates/volumes.libsonnet';
       },
     },
   },
+  TfNlpVisionMixin:: {
+    local config = self,
+    experiment: error 'Must define `experiment`',
+    additionalOverrides:: {},
+    configFiles: [],
```

Comment:
I wanted to have a minimal set of params within the top level, that way in the end tests we could avoid having to have boilerplate, e.g. comparing (1)
```
  local imagenet = {
    trainFilePattern: '$(IMAGENET_DIR)/train*',
    evalFilePattern: '$(IMAGENET_DIR)/valid*',
  },
```

with (2)
```
  local imagenet = {
    scriptConfig+: {
      trainFilePattern: '$(IMAGENET_DIR)/train*',
      evalFilePattern: '$(IMAGENET_DIR)/valid*',
    },
  },
```

I think the former is cleaner with the cost of having one time uglier code in the `common.libsonnet`. If anything I could just get rid of `scriptConfig` altogether but it's more convenient for the string formatting :)

Reply:
I take your point about boilerplate. In general, I want to keep new parameters that aren't common to all of the tests out of the top level as much as possible. It can already be hard to find where parameters in parent objects come from/where they are used, and nesting related groups of fields into objects can make that easier IMO. Also, most of the other inherited fields at the top level pertain to the test environment (e.g. benchmark ID, accelerators, volumes. timeout) and not the specific test

Reply:
Good point, I have moved everything into `scriptConfig`